mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/RabbitMQEventBusLifeCycleTest.java

```diff
@@ -0,0 +1,262 @@
+        public void beforeEach(ExtensionContext context) throws Exception {
+            rabbitMQExtension.beforeEach(context);
+
+            rabbitMQManagement = rabbitMQExtension.managementAPI();
+            connectionFactory = rabbitMQExtension.getConnectionFactory();
+
+            TestId.Factory mailboxIdFactory = new TestId.Factory();
+            routingKeyConverter = RoutingKeyConverter.forFactories(new MailboxIdRegistrationKey.Factory(mailboxIdFactory));
+            eventSerializer = new EventSerializer(mailboxIdFactory, new TestMessageId.Factory());
+        }
+
+        @Override
+        public void afterEach(ExtensionContext context) throws Exception {
+            rabbitMQExtension.afterEach(context);
+        }
+
+        private RabbitMQEventBus newEventBus() {
+            return new RabbitMQEventBus(connectionFactory, eventSerializer, routingKeyConverter);
+        }
+    }
+
+    private static class GroupA extends Group {}
+
+    static RabbitMQExtension rabbitMQExtension = new RabbitMQExtension();
+    @RegisterExtension
+    static RabbitMQEventBusDeclarationExtension testExtension = new RabbitMQEventBusDeclarationExtension();
+
+    @BeforeAll
+    static void setUp() {
+        rabbitMQExtension.beforeAll(null);
+    }
+
+    @AfterAll
+    static void tearDown() {
+        rabbitMQExtension.afterAll(null);
+    }
+
+    @Nested
+    class SingleEventBus {
+
+        private RabbitMQEventBus eventBus;
+
+        @BeforeEach
+        void setUp() {
+            eventBus = testExtension.newEventBus();
+        }
+
+        @AfterEach
+        void tearDown() {
+            eventBus.stop();
+        }
+
+        @Test
+        void startShouldCreateEventExchange() {
+            eventBus.start();
+            assertThat(testExtension.rabbitMQManagement.listExchanges())
+                .filteredOn(exchange -> exchange.getName().equals(MAILBOX_EVENT_EXCHANGE_NAME))
+                .hasOnlyOneElementSatisfying(exchange -> {
+                    assertThat(exchange.isDurable()).isTrue();
+                    assertThat(exchange.getType()).isEqualTo(DIRECT_EXCHANGE);
+                });
+        }
+
+        @Disabled("queue is declared with durable = true but management API always return durable = false. Suspecting a bug" +
+            " from rabbitmq driver")
+        @Test
+        void startShouldCreateKeyRegistrationWorkQueue() {
+            eventBus.start();
+            String registrationKeyWorkQueueName = eventBus.getKeyRegistrationHandler().getRegistrationQueue().asString();
+            assertThat(testExtension.rabbitMQManagement.listQueues())
+                .filteredOn(queue -> queue.getName().equals(registrationKeyWorkQueueName))
+                .hasOnlyOneElementSatisfying(queue -> {
+                    assertThat(queue.isDurable()).isTrue();
+                    assertThat(queue.isExclusive()).isTrue();
+                    assertThat(queue.isAutoDelete()).isTrue();
+                });
```

Comment:
While these tests are interesting they only cover technical implementation details, I do wonder the added value here, as it is not related to "features" and "behaviours" of the EventBus.